packages/astro/src/assets/fonts/utils.ts

```diff
@@ -15,26 +15,31 @@ import type {
 	ResolvedLocalFontFamily,
 } from './types.js';
 
-// Source: https://github.com/nuxt/fonts/blob/main/src/css/render.ts#L7-L21
-export function generateFontFace(family: string, font: unifont.FontFaceData) {
-	return [
-		'@font-face {',
-		`  font-family: ${family};`,
-		`  src: ${renderFontSrc(font.src)};`,
-		`  font-display: ${font.display ?? 'swap'};`,
-		font.unicodeRange && `  unicode-range: ${font.unicodeRange};`,
-		font.weight &&
-			`  font-weight: ${Array.isArray(font.weight) ? font.weight.join(' ') : font.weight};`,
-		font.style && `  font-style: ${font.style};`,
-		font.stretch && `  font-stretch: ${font.stretch};`,
-		font.featureSettings && `  font-feature-settings: ${font.featureSettings};`,
-		font.variationSettings && `  font-variation-settings: ${font.variationSettings};`,
-		`}`,
-	]
-		.filter(Boolean)
+export function toCSS(properties: Record<string, string | undefined>, indent = 2) {
+	return Object.entries(properties)
+		.filter(([, value]) => Boolean(value))
+		.map(([key, value]) => `${' '.repeat(indent)}${key}: ${value};`)
 		.join('\n');
 }
 
+export function renderFontFace(properties: Record<string, string | undefined>) {
+	return `@font-face {\n${toCSS(properties)}\n}\n`;
+}
+
+export function generateFontFace(family: string, font: unifont.FontFaceData) {
+	return renderFontFace({
+		'font-family': family,
+		src: renderFontSrc(font.src),
+		'font-display': font.display ?? 'swap',
+		'unicode-range': font.unicodeRange?.join(','),
+		'font-weight': Array.isArray(font.weight) ? font.weight.join(' ') : font.weight?.toString(),
```

Comment:
```suggestion
		'font-weight': Array.isArray(font.weight) ? font.weight.map(s => s.trim()).join(' ') : font.weight?.toString(),
```

nit: we might want to clean the string

Reply:
If we want to clean it up, it should happen earlier eg. in `resolveFamily`. But I think we don't want to clean it up, it's the responsability of font providers to return correct data